<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="DESKTOP-LDBB7HE" failures="0" tests="8" name="TestPackage.cartTestcases" time="26.951" errors="1" timestamp="2024-01-09T19:31:30 EET" skipped="1">
  <testcase classname="TestPackage.cartTestcases" name="verifyProductPriceValidDuringCheckoutProcess" time="0.296">
    <error message="Index 4 out of bounds for length 4" type="java.lang.IndexOutOfBoundsException">
      <![CDATA[java.lang.IndexOutOfBoundsException: Index 4 out of bounds for length 4
at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
at java.base/java.util.Objects.checkIndex(Objects.java:359)
at java.base/java.util.ArrayList.get(ArrayList.java:427)
at PageObjects.ProductsPage.getRandomProductPrice(ProductsPage.java:43)
at TestPackage.cartTestcases.verifyProductPriceValidDuringCheckoutProcess(cartTestcases.java:151)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:262)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.access$000(SuiteRunner.java:32)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:467)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
    </error>
  </testcase> <!-- verifyProductPriceValidDuringCheckoutProcess -->
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="verifyProductPriceValidDuringCheckoutProcess" time="0.322">
    <skipped/>
  </testcase> <!-- verifyProductPriceValidDuringCheckoutProcess -->
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="checkOutTheorder" time="6.260"/>
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="verifyTotalPrice" time="7.071"/>
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="fillCheckoutDetials" time="4.060"/>
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="verifyCartPageOpened" time="1.107"/>
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="removeIteamFromcart" time="6.497"/>
  <system-out/>
  <testcase classname="TestPackage.cartTestcases" name="verifyContinueShoppingButton" time="1.338"/>
  <system-out/>
</testsuite> <!-- TestPackage.cartTestcases -->
